tests/test_basic_ops_dict.py

```diff
@@ -398,17 +443,162 @@ def test_dict_pop_error(cfg: Dict[Any, Any], key: Any, expectation: Any) -> None
             "incompatible_key_type",
             False,
         ),
+        (
+            DictConfig(content={Enum1.FOO: "foo"}, key_type=Enum1, element_type=str),
+            "FOO",
+            True,
+        ),
+        (
+            DictConfig(content={Enum1.FOO: "foo"}, key_type=Enum1, element_type=str),
+            None,
+            False,
+        ),
+        # int key type
         ({1: "a", 2: {}}, 1, True),
         ({1: "a", 2: {}}, 2, True),
         ({1: "a", 2: {}}, 3, False),
         ({1: "a", 2: "???"}, 2, False),
+        ({1: "a", 2: "???"}, None, False),
+        ({1: "a", 2: "???"}, "1", False),
+        # (DictConfig({1: "a", 2: "???"}, key_type=int), "1", True),
+        # float key type
+        ({1.1: "a", 2.2: {}}, 1.1, True),
+        ({1.1: "a", 2.2: {}}, "1.1", False),
+        (DictConfig({1.1: "a", 2.2: {}}, key_type=float), "1.1", False),
+        ({1.1: "a", 2.2: {}}, 2.2, True),
+        ({1.1: "a", 2.2: {}}, 3.3, False),
+        ({1.1: "a", 2.2: "???"}, 2.2, False),
+        ({1.1: "a", 2.2: "???"}, None, False),
+        # bool key type
+        ({True: "a", False: {}}, True, True),
+        ({True: "a", False: {}}, False, True),
+        ({True: "a", False: {}}, "no", False),
+        ({True: "a", False: {}}, 1, True),
+        ({True: "a", False: {}}, None, False),
+        ({True: "a", False: "???"}, False, False),
     ],
 )
 def test_in_dict(conf: Any, key: str, expected: Any) -> None:
     conf = OmegaConf.create(conf)
     assert (key in conf) == expected
 
 
+@pytest.mark.parametrize(
```

Comment:
That's a pretty important addition.
1. Please put it in a different file.
2. Please document it a bit, there is a lot going on here.

Reply:
Done, in https://github.com/omry/omegaconf/pull/484/commits/49590a98dda3da265aaac7109a2440abf6f49c8f